docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,39 @@
+## Rebase OSE on Origin
+
+Today, code in OSE lands by fetching and merging Origin manually. OSE-specific changes
+are scattered in the repo and it's hard to reason about any conflicts. This doc proposes
+moving all the changes in OSE that diverge from Origin into a set of commits that
+will sit on top of the Origin code[0] and a Jenkins job that will be rebasing master and
+all active release branches on a daily basis.
+
+There are two types of commits that need to be carried on top of Origin: 
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs (release branches only)
+
+Rebases are going to be simpler for the master branch. The OSE master will be hard-reset to
+the Origin master and any carry commits will be applied on top. If any conflicts are hit, manual
+resolution will be required. In the future, we can extend the conflict resolution mechanism in
+Jenkins to send an e-mail either to the author and committer of the carry commit or we can store
+e-mails in the commit description and use those instead.
+
+Rebases for release branches are less trivial because they may contain backported fixes for CVEs
+that do not exist in the respective Origin branches. In such a case, backports should continue
+to be carried until we identify that the backport exists in the Origin branch and drop the OSE commit.
+How? Assuming the backport ends up w/o any additional change in Origin then cherry-picking
```

Comment:
[from @kargakis]
I didn't propose anywhere in the proposal using the "UPSTREAM: NUMBER: MESSAGE" pattern. I expected that the rebase would work like this:
(assuming we know the hashes of our carries / bugfixes)
```
$ git reset --hard upstream/release-1.*
$ git cherry-pick $commit_hash
```

If the cherry-pick succeeds, we can move to the next. If not, there are two reasons why it can fail:

1. either there are conflicts that need to be resolved or
2. a bugfix (even with a different commit message) with the exact diff cannot be applied because the diff already exists
    so you get something like

```
On branch test
You are currently cherry-picking commit 665673d.

nothing to commit, working directory clean
The previous cherry-pick is now empty, possibly due to conflict resolution.
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git reset'
```

Unfortunately I may end up parsing output in case of an error to separate between the two cases but at least this solves the naming problem.